theseus/optimizer/linear/lu_cuda_sparse_solver.py

```diff
@@ -0,0 +1,107 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import Any, Dict, List, Optional, Type
+
+import torch
+
+from theseus.core import Objective
+from theseus.extlib.cusolver_lu_solver import CusolverLUSolver
+from theseus.optimizer import Linearization, SparseLinearization
+from theseus.optimizer.autograd import LUCudaSolveFunction
+
+from .linear_solver import LinearSolver
+
+
+class LUCudaSparseSolver(LinearSolver):
+    def __init__(
+        self,
+        objective: Objective,
+        linearization_cls: Optional[Type[Linearization]] = None,
+        linearization_kwargs: Optional[Dict[str, Any]] = None,
+        damping: float = 1e-6,
+        batch_size: int = 16,
+        num_solver_contexts=1,
+        **kwargs,
+    ):
+        linearization_cls = linearization_cls or SparseLinearization
+        if not linearization_cls == SparseLinearization:
+            raise RuntimeError(
+                "LUCudaSparseSolver only works with theseus.optimizer.SparseLinearization,"
+                + f" got {type(self.linearization)}"
+            )
+
+        super().__init__(objective, linearization_cls, linearization_kwargs, **kwargs)
+        self.linearization: SparseLinearization = self.linearization
+
+        if self.linearization.structure().num_rows:
+            self.reset()
+
+        # the `damping` has the purpose of (optionally) improving conditioning
+        # self._damping: float = damping
```

Comment:
Is LM not supported at the moment?

Reply:
dead code, the parameter is set in solve, will be cleaned.